ml-proto/spec/int.ml

```diff
@@ -208,7 +208,35 @@ struct
   let ge_s x y = x >= y
   let ge_u x y = cmp_u x (>=) y
 
-  let of_string = Rep.of_string
+  (* to_string supports unsigned decimals *)
+  let of_string x =
+    let unsigned_decimal x =
+      let ten = Rep.of_int 10 in
+      let l = String.length x in
+      (* First overflow-check via String.length *)
+      if l > (if Rep.bitwidth = 32 then 10 else 20)
+        then raise (Failure "int_to_string");
+      let firstPart = Rep.of_string (String.sub x 0 (l - 1)) in
+      let secondPart = Rep.of_string (String.sub x (l - 1) 1) in
+      (* Negativity-check *)
+      if firstPart < Rep.zero
+        then raise (Failure "int_to_string");
+      (* Second overflow-check *)
+      if firstPart > (div_u Rep.minus_one ten)
+      || (firstPart = (div_u Rep.minus_one ten) && secondPart > (rem_u Rep.minus_one ten))
+        then raise (Failure "int_to_string");
+      Rep.add
```

Comment:
Nit: can we put this on one line? Also, you can shorten to `Rep.(add (mul ...) ...)`